apps/tup-ui/src/auth/LoginComponent.tsx

```diff
@@ -12,29 +12,27 @@ export const LogoutComponent: React.FC = () => {
 };
 
 const LoginComponent: React.FC = () => {
-  const { login, loggedIn, error, isLoading } = useAuth();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const from = (location.state as { from?: Location })?.from?.pathname || '/';
+  const authCallback = useCallback(
+    () => navigate(from, { replace: true }),
```

Comment:
Any way we can set the navigation destination as an optional property of LoginComponent?

Reply:
I don't understand what this would buy us. Passing the destination as a prop would split the redirect logic between `LoginComponent` and its parent, which seems like complexity/coupling we don't want.

Reply:
Oh, gotcha. I just had designs and visions of custom elements that allow capture of the user's token into the cookie for any consumer of the component, be it react or something else.

Reply:
I also realize we can't do a redirect outside of the react application without doing window.location, which would break the router.

Reply:
Okay I think i know what i was trying to get at here:  ` || '/'` amounts to a hardcoded default login redirect target should no "from" exist. That's probably what I meant by navigation destination. We don't have to fix this right now.